docs/features/software-catalog/references.md

```diff
@@ -0,0 +1,104 @@
+help the writer, the catalog has a few tricks up its sleeve.
+
+Each reference can be expressed in one of two ways: as a compact string, or as a
+compound reference structure.
+
+## String References
+
+This is the most common alternative, that should be used in almost all
+circumstances.
+
+The string is on the form `[<kind>:][<namespace>/]<name>`, that is, it is
+composed of between one and three parts in this specific order, without any
+additional encoding:
+
+- Optionally, the kind, followed by a colon
+- Optionally, the namespace, followed by a forward slash
+- The name
+
+The name is always required. Depending on the context, you may be able to leave
+out the kind and/or namespace. If you do, it is contextual what values will be
+used, and the relevant documentation should specify which rule applies where.
+All strings are case insensitive.
+
+```yaml
+# Example:
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: petstore
+  namespace: external-systems
+  description: Petstore
+spec:
+  type: service
+  lifecycle: experimental
+  owner: group:pet-managers
+  implementsApis:
+    - petstore
+    - internal/streetlights
+    - hello-world
+```
+
+The field `spec.owner` is a reference. In this case, the string
+`group:pet-managers` was given by the user. That means that the kind is `Group`,
+the namespace is left out, and the name is `pet-managers`. In this context, the
+namespace was chosen to fall back to the value `default` by the code that parsed
+the reference, so the end result is that we expect to find another entity in the
+catalog that is of kind `Group`, namespace `default` (which, actually, also can
+be left out in its own yaml file because that's the default value there too),
+and name `pet-managers`.
+
+The entries in `implementsApis` are also references. In this case, none of them
+need to specify a kind since we know from the context that that's the only kind
+that's supported here. The second entry specifies a namespace but the other ones
+don't, and in this context, the default is to refer to the same namespace as the
+originating entity (`external-systems` here). So the three references
+essentially expand to `api:external-systems/petstore`,
+`api:internal/streetlights`, and `api:external-systems/hello-world`. We expect
+there to exist three API kind entities in the catalog matching those references.
+
+## Compound References
+
+This is a more verbose version of a reference, where each part of the
+kind-namespace-name triplet is expressed as a field in a structure. This format
+can be used where necessary, such as if either of the three elements contain
+colons or forward slashes. Avoid using it where possible, since it is harder to
+read and write for humans.
+
+```yaml
+# Example:
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: petstore
+  description: Petstore
+spec:
+  type: service
+  lifecycle: experimental
+  owner:
+    kind: Group
+    name: aegis-imports/pet-managers
```

Comment:
Is the actual name of the component here `aegis-imports/pet-managers`? Thought we didn't allow `/` in component names

Reply:
Just a theoretical example :) Yes this should be rare

Reply:
No I mean we shouldn't accept this as input? It can exist in the file, but we would make `parseEntityRef` or w/e throw?

Reply:
Not really. The rule of "no slashes" is not fixed, it's configurable. It's perfectly possible that an org makes a custom validator for their entities and allows slashes and/or colons.

Reply:
Seems like a lot of complication for little value to allow that to be configurable

Reply:
One day we get that customer that goes "oh but our existing inventory and external systems has slashes all over the place, what do you mean you chose to hard code not supporting that?" :P it kinda grinds me to even consider deciding on a thing like that, usually it turns out to not be done for particularly good reason :)

Reply:
(being a bit stoic here, i know, but that's how it's built already actually)

Reply:
I don't think the plugin ecosystem is gonna support that though, not unless we're way more clear that that is the case

Reply:
I think it's just up to basic sound engineering practices right? Like encodeURIComponent when building urls etc

Reply:
Yeah but that's not done throughout the codebase atm